src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const negativeLength = sides.some((side) => side <= 0);
+    const sumOfSides = sides.reduce((sum, side) => sum + side, 0);
+    const biggestSide = Math.max(...sides);
+    const isBiggest = biggestSide >= sumOfSides - biggestSide;
+
+    if (negativeLength) {
+      throw new Error('One of the sides is negative');
+    }
```

Comment:
you need to check that input arguments are bigger than 0 for each figure
instead of checking it in each class, let's create one separate function for this purpose, and let's call it for each figure